### DISCUSSION TITLE
JuliaSIMD ecosystem deprecated for Julia 1.11 and above

### DISCUSSION THREAD

**Discussion by verseve** [Deltares/Wflow.jl#361] [RFC]
Wflow relies on two packages from the JuliaSIMD ecosystem:
- [LoopVectorization](https://github.com/JuliaSIMD/LoopVectorization.jl)
- [Polyester](https://github.com/JuliaSIMD/Polyester.jl)

It is announced that "Without new maintainers for the JuliaSIMD ecosystem, LoopVectorization.jl and [JuliaSIMD](https://github.com/JuliaSIMD) will be deprecated for Julia v1.11 and above" on LoopVectorization github page.

See also for more info:
https://github.com/JuliaSIMD/LoopVectorization.jl/pull/519
https://discourse.julialang.org/t/why-is-loopvectorization-deprecated/109547

Discussion and decision at some point is needed, especially if new maintainers for JuliaSIMD are not available.

**Comment by visr** [comment]
Yeah it's good to keep an eye on this. I think these packages achieve nice speedups, but since they rely on Julia internals they are quite brittle and hard to maintain. This is also why the main author is working on [`LoopModels`](https://github.com/LoopModels/LoopModels) to implement the optimizations at the LLVM level.

If we run into more issues like #360 we could decide to drop it already, and hope we'll get the speed back with future compiler optimizations. But for now I'd say let's wait it out a bit.

**Comment by visr** [comment]
I see the PackageCompiler build is currently broken on master, and #514 also doesn't help. The errors look like this:

```
- PackageCompiler: compiling nonincremental system image
13:25:35   LLVM ERROR: Cannot select: 0x7f02e99f6e20: v2f64 = X86ISD::VGETMANT 0x7f02e9a25b58, TargetConstant:i32<2>, /u/svc-teamcity-ansible/.julia/packages/VectorizationBase/wHnQd/src/special/exp.jl:906
```

Considering https://github.com/JuliaSIMD/LoopVectorization.jl/issues/432#issuecomment-1252718107 we may want to go ahead with removing the LoopVectorization.jl dependency before v1.0. I think Polyester is fine and can stay, despite it being in the JuliaSIMD org.

**Reply by vers-w** [reply]
Hm, strange that the PackageCompiler build throws an error on the build server. This is as far as I know actually the first time (related to `LoopVectorization`) on the build server and I cannot reproduce this issue on my laptop. Do you perhaps know if there were any changes recently on the build server?

**Reply by visr** [reply]
Not that I know. Indeed strange that this suddenly started happening.
